app/controllers/PaypalController.scala

```diff
@@ -0,0 +1,103 @@
+package controllers
+
+import actions.CommonActions._
+import com.gu.i18n.CountryGroup
+import play.api.data.{Form, FormError}
+import play.api.data.Forms._
+import play.api.libs.ws.WSClient
+import play.api.mvc.{Controller, Result}
+import services.PaymentServices
+import play.api.Logger
+import play.api.data.format.Formatter
+import play.api.libs.json._
+import utils.ContributionIdGenerator
+import play.api.libs.functional.syntax._
+
+import scala.util.Right
+
+
+class PaypalController(ws: WSClient, paymentServices: PaymentServices, contributionIdGenerator :ContributionIdGenerator) extends Controller {
```

Comment:
I'm being pedantic but the spacing for the parameter contributionIdGenerator is wrong